### PR TITLE
Add support for BigInt in all amount values

### PR DESCRIPTION
**Pull request by ratik21**
Added bigint support for: transfer amount for assets and algo, asset issueance

NOTES:
+ Since the [pull req](https://github.com/algorand/js-algorand-sdk/pull/263) to support bigint is not released yet by `algosdk`, to upgrade the package, i updated it by the latest commit (using `yarn add <git_url>#<commit_hash>`). 
+ Also, algosdk REST API's does not support `bigint` in their response yet, so the accountInfo api (eg. `algod.accountInformation(account).do())` will lose precision with amount. I have tested the transactions with `bigint` using `goal` and added notes in code. This is the open PR for this support: [https://github.com/algorand/js-algorand-sdk/pull/260](https://github.com/algorand/js-algorand-sdk/pull/260)

UPDATE - Ready for Review

### CODE REVIEW

File: examples/asa/package.json
Comment (robert-zaremba) [inline examples/asa/package.json:10]: Please use the syntax described in this article: https://medium.com/@jonchurch/use-github-branch-as-dependency-in-package-json-5eb609c81f1a

File: examples/crowdfunding/package.json
Comment (robert-zaremba) [inline examples/crowdfunding/package.json:7]: as above

File: packages/algob/src/lib/algo-operator.ts
Comment (robert-zaremba) [inline packages/algob/src/lib/algo-operator.ts:96]:
```suggestion
    return BigInt(accoutInfo.amount) - BigInt(accoutInfo.assets.length + 1) * ALGORAND_ASA_OWNERSHIP_COST;
```

File: packages/runtime/src/account.ts
Comment (robert-zaremba) [inline packages/runtime/src/account.ts:358]: what is this comment?

**Review by robert-zaremba — COMMENTED** [review]
Please update CHANGELOG as well

File: examples/asa/package.json
Comment (ratik21) [reply examples/asa/package.json:10]: Okay, thanks for suggestion

File: packages/runtime/src/account.ts
Comment (ratik21) [reply packages/runtime/src/account.ts:358]: Sorry, that shouldn't be here :sweat_smile:

File: examples/asa/package.json
Comment (ratik21) [reply examples/asa/package.json:10]: Updated to `user/repo#commit` (`algorand/js-algorand-sdk.git#9a5824782511dff1f521c68fe54d562b3872d1ae`)

File: examples/crowdfunding/package.json
Comment (ratik21) [reply examples/crowdfunding/package.json:7]: done

File: CHANGELOG.md
Comment (robert-zaremba) [inline CHANGELOG.md:13]:
```suggestion
* Use `bigint` for all numeric values in `runtime` and `algob` to support integers upto 64 bit(`uint64`).
```

**Review by robert-zaremba — APPROVED** [review]
left one suggestion. Please wait for @man-of-code  review before merging.

File: packages/types-algosdk/types/index.d.ts
Comment (amityadav0) [inline packages/types-algosdk/types/index.d.ts:222]: sd

I think this is not optional.

File: packages/runtime/src/types-input.ts
Comment (amityadav0) [inline packages/runtime/src/types-input.ts:26]: will this be `smaller than 2^64` ?

File: packages/algob/src/lib/script-checkpoints.ts
Comment (amityadav0) [inline packages/algob/src/lib/script-checkpoints.ts:151]: why are we doing this ?

File: packages/algob/src/internal/deployer.ts
Comment (ratik21) [reply packages/algob/src/internal/deployer.ts:92]: The latest `algosdk` code threw error that signature part of lsig should be a `Uint8Array` while processing txs.

File: packages/runtime/test/src/lib/asa.ts
Comment (ratik21) [reply packages/runtime/test/src/lib/asa.ts:92]: It's actually `2^64 - 1`. Updating.

File: packages/runtime/src/types-input.ts
Comment (ratik21) [reply packages/runtime/src/types-input.ts:26]: [https://github.com/algorand/js-algorand-sdk/blob/develop/src/transaction.js#L61](https://github.com/algorand/js-algorand-sdk/blob/develop/src/transaction.js#L61)

File: packages/algob/src/lib/script-checkpoints.ts
Comment (ratik21) [reply packages/algob/src/lib/script-checkpoints.ts:151]: Previously tag was being decoded in `cp` as `{ type: 'buffer', value: <value> }`. But we needed it to be `Uint8Array`, example projects were failing otherwise (on my system). You can check on your system too. Let me know if we don't need this.

File: packages/types-algosdk/types/index.d.ts
Comment (ratik21) [reply packages/types-algosdk/types/index.d.ts:222]: Yes. Updating `runtime/src/logicsig.ts` as well to add `tag`.

File: packages/algob/src/internal/deployer.ts
Comment (amityadav0) [reply packages/algob/src/internal/deployer.ts:92]: ok

File: packages/runtime/src/types-input.ts
Comment (amityadav0) [reply packages/runtime/src/types-input.ts:26]: `assetTotal <= 0xFFFFFFFFFFFFFFFFn`
It has equal though. 0x...n this is equal to 2^64 - 1 right ?

File: packages/runtime/test/src/lib/asa.ts
Comment (amityadav0) [reply packages/runtime/test/src/lib/asa.ts]: https://github.com/scale-it/algorand-builder/pull/184#discussion_r585808970

File: packages/algob/src/lib/script-checkpoints.ts
Comment (amityadav0) [reply packages/algob/src/lib/script-checkpoints.ts:151]: ok, we need this. Please add a comment here.(for later understanding)

Comment (amityadav0) [reply packages/algob/src/lib/script-checkpoints.ts:151]: can we do this before `registerLsig` ?

File: packages/runtime/src/types-input.ts
Comment (ratik21) [reply packages/runtime/src/types-input.ts:26]: Right. So i think the message should be `Total must be a positive number and smaller than or equal to 2^64-1`. And the message here seems incorrect then - [https://github.com/algorand/js-algorand-sdk/blob/develop/src/transaction.js#L61](https://github.com/algorand/js-algorand-sdk/blob/develop/src/transaction.js#L61)

File: packages/algob/src/lib/script-checkpoints.ts
Comment (ratik21) [reply packages/algob/src/lib/script-checkpoints.ts:151]: ok

File: packages/runtime/src/types-input.ts
Comment (amityadav0) [reply packages/runtime/src/types-input.ts:26]: yes or `smaller than 2^64`

File: packages/algob/src/lib/script-checkpoints.ts
Comment (ratik21) [reply packages/algob/src/lib/script-checkpoints.ts:151]: Yes. How about moving this to [`getLsig`](https://github.com/scale-it/algorand-builder/blob/master/packages/algob/src/lib/lsig.ts#L15) ?

**Review by man-of-code — APPROVED** [review]

File: packages/algob/src/lib/script-checkpoints.ts
Comment (amityadav0) [reply packages/algob/src/lib/script-checkpoints.ts:151]: yes that would be better or `mkDelegatedLsig` works too

File: packages/runtime/test/src/lib/asa.ts
Comment (amityadav0) [inline packages/runtime/test/src/lib/asa.ts:92]:
```suggestion
  it("Should check total to be a positive number < 2^64", async () => {
```

File: packages/algob/src/lib/script-checkpoints.ts
Comment (ratik21) [reply packages/algob/src/lib/script-checkpoints.ts:151]: Ok. moving to `getLsig`

Comment (amityadav0) [reply packages/algob/src/lib/script-checkpoints.ts:151]: examples are working on my system if we change tag to type `Buffer`. tested asa example

Comment (amityadav0) [reply packages/algob/src/lib/script-checkpoints.ts:151]: ** in types-algosdk

Comment (amityadav0) [reply packages/algob/src/lib/script-checkpoints.ts:151]: update: got error with buffer.

**Review by amityadav0 — APPROVED** [review]

**Pull request merged**
